docs/architecture/017-applicant-ux.md

```diff
@@ -0,0 +1,66 @@
+# 017: Applicant Form UX
+
+Authors: @opqdonut, @Macroz, @marharyta
+
+This ADR aims to collect together some topics like validations in
+frontend, autosaving, etc. that are related to the Applicant's user
+experience. There have been previous discussions about all of these
+topics but no unified presentation.
+
+Based on ticket [#2639](https://github.com/CSCfi/rems/issues/2639)
+
+# Problems with the current applicant UX
+
+## A) Validation errors block saving
+
+You don't know if your inputs (e.g. ip address) are valid before
+pressing Save, and then you need to fix all of these errors before
+REMS saves your work.
+
+This can be irritating if you're working on a long application, and
+just want to save your work and go home. It's less of an issue if
+users save early & often.
+
+**Possible solution**: Show validation errors to the user, but perform
```

Comment:
Could mention that it used to work like this?

Reply:
I started doing that and tried to dig out the PR that changed things, but couldn't find a concrete decision to do this. I'll write my findings down.